<a href="https://colab.research.google.com/github/JuanDb20/Matematicas-Aplicadas-1/blob/main/Taller1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# <font color="#401D87">***Taller #1*** </font>

Presentado por: <font color="#9D84CF">Isabella Candado Conde, Juan Diego Balanta Molina, Edward Estacio, Juan Pablo Martinez Rosero </font>

## <font color="#6E579E">**1. Primer punto** </font>

---
<font color="#9D84CF">**1.1.** </font> Resolver los polinomios aritméticos:
>* $\{[3(-5+10-5)]+[-2(-2-4+10)]\}^2 = 64$

In [ ]:
ej1_1=int((3*((-5)+10+(-5)))+((-2)*((-2)+(-4)+10))**2)
print (ej1_1)

64


>* $45+[3(2+8)-\frac{8}{4}]+10-83 = 0$

In [ ]:
ej1_2=int(45+((3*(2+8))-(8/4))+10-83)
print (ej1_2)

0


>* $[2(-4+7)-10]*(-4-7-3)= 56$

In [ ]:
ej1_3=int(((2*(-4+7))-10)*(-4-7-3))
print (ej1_3)

56


---
<font color="#9D84CF">**1.2.** </font> Resolver las siguientes combinaciones de operaciones:

>* $\frac{1}{5}[(\frac{1}{2})^2 + (\frac{3}{4})^2]+\frac{10}{25}[(\frac{1}{2})^2+(\frac{3}{4})^2]+\frac{8}{40}[(\frac{1}{2})^2 + (\frac{3}{4})^2] = 0$

In [ ]:
ej2_1=int(((1/5)*((1/2)**2 + (3/4)**2))+((2/5)*((1/2)**2 + (3/4)**2))+((1/5)*((1/2)**2 + (3/4)**2)))
print (ej2_1)

0


>* $\frac{9}{4}(\frac{6}{8})^{-1}-\frac{5}{2}(\frac{25}{20})^{-1}+\frac{40}{8}(\frac{20}{16})^{-1} = 5$

In [ ]:
ej2_2=int(((9/4)*(3/4)**-1)-((5/2)*(5/4)**-1)+((5)*(5/4)**-1))
print (ej2_2)

5


---
## <font color="#6E579E">**2. Segundo punto** </font>


<font color="#9D84CF">**2.1.** </font> Despejar la variable indicada en cada una de las siguientes funciones:

|N°|Función|Variable a despejar|Formulas Matemáticas|Función despejada|
|--|--|--|--|--|
|1|Área de un cilindro|$r$|$A=2πrg$|$r=\frac{A}{2πg}$|
|2|Volumen del cono|$h$|$V=\frac{A_B*h}{3}$|$h=\frac{3V}{A_B}$|
|3|Volumen de la esfera|$r$|$V=\frac{4}{3}*π*r^3$|$r=\sqrt[3]{\frac{3V}{4π}}$|
|4|Área del trapecio|$h$|$A=\frac{B+b}{2}*h$|$h=\frac{2A}{B+b}$
|5|Intensidad Eléctrica|$t$|$I=\frac{q}{t}$|$t=\frac{q}{I}$
|6|Ley de Ohm|$R$|$I=\frac{V}{R}$|$R=\frac{V}{I}$
|7|Energía Eléctrica|$t$|$E=\frac{V^2}{R}*t$|$t=\frac{E*R}{V^2}$
|8|Resistencias en paralelo|$R$|$\frac{1}{R}=\frac{1}{R_1}+\frac{1}{R_2}$|$R=\frac{R_1*R_2}{R_1+R_2}$
|9|Ecuación gases perfectos|$P_f$|$\frac{P_0 V_0}{T_0}=\frac{P_f V_f}{T_f}$|$P_f=\frac{T_f P_0 V_0}{T_0 V_f}$
|10|Intensidad Eléctrica|$V_B$|$I=\frac{V_A - V_B}{R}$|$V_B=V_A - IR$

---
<font color="#9D84CF">**2.2.** </font> Prueba con valores aleatorios las siguientes funciones a trozos:

> $f(x)=\left\{\begin{array}{lcc}x+2 & si & x < -3 \\ \\ -x & si & -3 \leq x < 0 \\ \\ 2 & si & x \geq 0 \end{array} \right.$

In [ ]:
val1=[-4,-3,-2,-1,0,1]
for x in val1:
    if x<-3:
        print (x+2)
    elif -3<=x<0:
        print (-x)
    else:
        print (2)

-2
3
2
1
2
2


>$f(x)=\left\{\begin{array}{lcc}a-x^2 & si & x < 0 \\ \\ 2 & si & 0 \leq x \leq 1 \\ \\ x^2-2x+b & si & x > 1 \end{array} \right.$

In [ ]:
val2=[-1,0,1,2]
a=int(input("Ingrese el valor de a: "))
b=int(input("Ingrese el valor de b: "))
for x in val2:
    if x<0:
        print (a-x**2)
    elif 0<=x<=1:
        print (2)
    else:
        print (x**2-2*x+b)

Ingrese el valor de a: 4
Ingrese el valor de b: 8
3
2
2
8


---
## <font color="#6E579E">**3. Tercer punto** </font>


<font color="#9D84CF">**3.1.** </font> Escribir una función que reciba una muestra de números en una lista y devuelva otra lista con sus cuadrados, otra con los cubos y otra con las
raices:

In [ ]:
def calculo_potencias_y_raices():
    lista_num = []
    for i in range(5):
        numero = float(input(f"Ingresa el número {i+1}: "))
        lista_num.append(numero)

    cuadrados = [x**2 for x in lista_num]
    cubos = [x**3 for x in lista_num]
    raices = [x**0.5 for x in lista_num]

    print("\nResultados:")
    print("Cuadrados:", cuadrados)
    print("Cubos:", cubos)
    print("Raices:", raices)

calculo_potencias_y_raices()

Ingresa el número 1: 2
Ingresa el número 2: 6
Ingresa el número 3: 5
Ingresa el número 4: 9
Ingresa el número 5: 7

Resultados:
Cuadrados: [4.0, 36.0, 25.0, 81.0, 49.0]
Cubos: [8.0, 216.0, 125.0, 729.0, 343.0]
Raices: [1.4142135623730951, 2.449489742783178, 2.23606797749979, 3.0, 2.6457513110645907]


---
<font color="#9D84CF">**3.2.** </font> Escribir un programa que almacene las asignaturas vistas el semestre pasado en una lista, pregunte al usuario la nota que ha sacado en cada asignatura. Haga un print donde muestre al usuario la siguiente información:  
* Si la nota es Mayor o igual a 4 debe mostrar un mensaje que diga:
”Nivel superior”.
* Si la nota esta entre 3,5 y 4 (menor a 4), debe mostrar un mensaje que diga: "Asignatura aprobada pero debe asistir a plan de mejoramiento,
dirijase a CAMBAS".
* Si la nota es menor a 3,5 debe aparecer: "Asignatura no aprobada, debe
repetirla el próximo semestre".
  * Punto adicional. Haga un programa que almacene cada asignatura
en una tabla con su respectiva nota y el mensaje asociado a la nota.

In [ ]:
num_asig = int(input("Ingrese el número de asignaturas: "))
tabla = [[],[],[]]

for i in range(num_asig):
    asig = str(input(f"\nNombre de la asignatura {i+1}: "))
    tabla[0].append(asig)
    nota = float(input(f"Nota de la asignatura {asig}: "))
    tabla[1].append(nota)

    if nota >= 4:
        msj = "Nivel superior"
        tabla[2].append(msj)
        print(msj)
    elif 3.5 <= nota < 4:
        msj = "Asignatura aprobada pero debe asistir a plan de mejoramiento, dirijase a CAMBAS"
        tabla[2].append(msj)
        print(msj)
    else:
        msj = "Asignatura no aprobada, debe repetirla el próximo semestre"
        tabla[2].append(msj)
        print(msj)

print("\nTabla de notas:")
print("Asignatura\t\tNota\tMensaje")

for i in range(num_asig):
    print(f"{tabla[0][i]}\t\t\t{tabla[1][i]}\t{tabla[2][i]}")

Ingrese el número de asignaturas: 4

Nombre de la asignatura 1: LyA
Nota de la asignatura LyA: 2.5
Asignatura no aprobada, debe repetirla el próximo semestre

Nombre de la asignatura 2: AyF
Nota de la asignatura AyF: 3.8
Asignatura aprobada pero debe asistir a plan de mejoramiento, dirijase a CAMBAS

Nombre de la asignatura 3: COE
Nota de la asignatura COE: 4.7
Nivel superior

Nombre de la asignatura 4: FDC
Nota de la asignatura FDC: 3
Asignatura no aprobada, debe repetirla el próximo semestre

Tabla de notas:
Asignatura		Nota	Mensaje
LyA			2.5	Asignatura no aprobada, debe repetirla el próximo semestre
AyF			3.8	Asignatura aprobada pero debe asistir a plan de mejoramiento, dirijase a CAMBAS
COE			4.7	Nivel superior
FDC			3.0	Asignatura no aprobada, debe repetirla el próximo semestre


---
<font color="#9D84CF">**3.3.** </font> Construya una lista que almacene el precio del barril de Petróleo los ultimos 7 días. Indague sobre el precio de un litro de Coca-Cola (Hoy) y haga la conversión de litros a barriles. Construya otra lista que relacione proporcionalemente el precio de un barril de Petróleo vs un barril de Coca-Cola. Construya un párrafo con sus conclusiones.


In [ ]:
precio_petroleo = [76.75, 75.34, 75.03, 78.85, 80.79, 80.65, 79.58]

precio_litro_coca = 0.80

litro_por_barril = 159  # Un barril equivale a aproximadamente 159 litros
precio_barril_coca = precio_litro_coca * litro_por_barril

relacion_precio1 = [precio_petroleo / precio_barril_coca for precio_petroleo in precio_petroleo]
relacion_precio2 = [precio_barril_coca / precio_petroleo for precio_petroleo in precio_petroleo]

print("Precios del barril de petróleo en los últimos 7 días (USD):", precio_petroleo)
print("Precio de un litro de Coca-Cola (USD):", precio_litro_coca)
print("Precio de un barril de Coca-Cola (USD):", precio_barril_coca)
print("Relación del precio del barril de petróleo respecto al precio del barril de Coca-Cola:", relacion_precio1)
print("Relación del precio del barril de Coca-Cola respecto al precio del barril de petróleo:", relacion_precio2)



Precios del barril de petróleo en los últimos 7 días (USD): [76.75, 75.34, 75.03, 78.85, 80.79, 80.65, 79.58]
Precio de un litro de Coca-Cola (USD): 0.8
Precio de un barril de Coca-Cola (USD): 127.2
Relación del precio del barril de petróleo respecto al precio del barril de Coca-Cola: [0.6033805031446541, 0.5922955974842767, 0.5898584905660378, 0.6198899371069182, 0.6351415094339623, 0.6340408805031447, 0.6256289308176101]
Relación del precio del barril de Coca-Cola respecto al precio del barril de petróleo: [1.657328990228013, 1.6883461640562782, 1.6953218712514995, 1.6131896005072925, 1.5744522836984773, 1.5771853688778672, 1.5983915556672532]


<font color="#9D84CF"> *Conclusiones:* </font>
<font color="807B84">
* La lista de precios del barril de petróleo muestra las variaciones diarias en los costos del petróleo durante la última semana.
* El precio actual de un litro de Coca-Cola se ha utilizado para calcular el costo de un barril de Coca-Cola.
* La división entre el precio de un barril de petróleo y un barril de   Coca-Cola muestra la relación entre estos dos bienes. En general, podemos observar que el barril de petróleo es considerablemente más barato en comparación con el barril de Coca-Cola. Lo cual es curioso, porque el petróleo es crucial en la industria y representa el motor principal de la economía mundial; sin embargo, su precio es menor a una bebida gasificada.
* Los resultados indican que:
  * El precio del barril de Coca-Cola es 0.6 veces más alto que el precio del barril del petróleo. Es decir, con el dinero que se compra un barril de petróleo se pueden comprar 0.6 barriles de Coca-Cola.
  * El precio del barril de Petróleo es 1.7 veces más barato que el precio del barril de Coca-Cola. Así que, con el dinero que se compra un barril de Coca-Cola se pueden comprar 1.7 barriles de Petróleo.
  </font>